lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableLink.cs

```diff
@@ -174,15 +174,16 @@ public T Read()
                 return default(T);
             }
 
-            T value = _cache.GetInstance(dataType);
-
-            if (value == null)
+            try
+            {
+                T value = (T) _injector.ForkInjector().GetInstance(dataType);
```

Comment:
.GetType() always give class type an not interface type

On Wed, May 13, 2015 at 11:43 AM, Julia Wang <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableLink.cs
> <https://github.com/apache/incubator-reef/pull/181#discussion_r30262383>:
>
> > @@ -174,15 +174,16 @@ public T Read()
> >                  return default(T);
> >              }
> >
> > -            T value = _cache.GetInstance(dataType);
> > -
> > -            if (value == null)
> > +            try
> > +            {
> > +                T value = (T) _injector.ForkInjector().GetInstance(dataType);
>
> How do you guarantee the dataType is a interface or a class?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/181/files#r30262383>.
>